### PR TITLE
added interfaces and read from supabase

### PR DESCRIPTION
**Pull request by vladutdinu**

### CODE REVIEW

File: src/lib/interfaces/course.interface.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { ISkill } from "./skill.interface";
+
+export interface ICourse {
+	id: string;
+	content: string;
+	subdomain_id: string;
```
Comment (smetytech) [inline src/lib/interfaces/course.interface.ts:6]: Please remove the `subdomain_id`

File: src/lib/interfaces/thread.interface.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export interface IThreadMessage {
+	id: string;
+	message: string;
+	thread_id: string;
+	user_id: string
```
Comment (smetytech) [inline src/lib/interfaces/thread.interface.ts:5]: Please remove the `_id` properties and maybe we can return some data about the user that wrote the message, like:
`user: { display_name: string }`

File: src/lib/interfaces/thread.interface.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export interface IThreadMessage {
+	id: string;
```
Comment (smetytech) [inline src/lib/interfaces/thread.interface.ts:2]: Maybe we should also add the `created_at` or `updated_at` property so we can display this information in the UI

File: src/lib/interfaces/thread.interface.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+	thread_id: string;
+	user_id: string
+}
+
+export interface IThread {
+	id: string;
+	name: string;
+	domain_id: string;
+	subdomain_id: string;
+	course_id: string;
```
Comment (smetytech) [inline src/lib/interfaces/thread.interface.ts:13]: Please remove the `_id` properties

File: src/routes/api/domains/+page.svelte [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import type { IDomain } from '$lib/interfaces/domain.interface';
+	let domains: IDomain[] = $state([]);
```
Comment (smetytech) [inline src/routes/api/domains/+page.svelte:4]: Please use the svelte 4 mode of work

File: src/routes/api/domains/+page.svelte [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+	// Fetch the domains from the API route
+	const fetchdomains = async () => {
+		const res = await fetch('/api/domains');
+		domains = await res.json();
+	};
```
Comment (smetytech) [inline src/routes/api/domains/+page.svelte:10]: Please move this into the `src/lib/controllers/domain.controller.ts` file

File: src/routes/api/domains/+page.svelte [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+	// Load the domains when the component mounts
+	onMount(() => {
+		fetchdomains();
```
Comment (smetytech) [inline src/routes/api/domains/+page.svelte:14]: Please use camelCase here

File: src/routes/api/domains/[id]/subdomains/+page.svelte [hunk 8]
```diff
@@ -0,0 +1,31 @@
+<script lang="ts">
+    import { onMount } from 'svelte';
+	  import type { ISubdomain } from "$lib/interfaces/subdomain.interface";
+
+  let subdomains: ISubdomain[] = $state([]);
```
Comment (smetytech) [inline src/routes/api/domains/[id]/subdomains/+page.svelte:5]: Again, svelte 4 please

File: src/routes/api/domains/[id]/subdomains/+page.svelte [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+  // Fetch the subdomains from the API route
+  const fetchsubdomains = async () => {
```
Comment (smetytech) [inline src/routes/api/domains/[id]/subdomains/+page.svelte:8]: camelCase and move the function into a controller folder

File: src/routes/api/domains/[id]/subdomains/[id]/courses/+page.svelte [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  // Fetch the courses from the API route
+  const fetchcourses = async () => {
+    const res = await fetch('/api/courses');
+    courses = await res.json();
+  };
+
+  // Load the courses when the component mounts
+  onMount(() => {
+    fetchcourses();
+  });
```
Comment (smetytech) [inline src/routes/api/domains/[id]/subdomains/[id]/courses/+page.svelte:15]: camelCase and controller folder please

File: src/routes/api/threads/+page.svelte [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  // Fetch the threads from the API route
+  const fetchTreads = async () => {
+    const res = await fetch('/api/threads');
+    threads = await res.json();
+  };
+
+  // Load the threads when the component mounts
+  onMount(() => {
+    fetchTreads();
+  });
```
Comment (smetytech) [inline src/routes/api/threads/+page.svelte:16]: Same here, camelCase + controller folder

**Review by smetytech — CHANGES_REQUESTED** [review]

File: src/lib/interfaces/course.interface.ts (hunk 1 above)
Comment (vladutdinu) [reply src/lib/interfaces/course.interface.ts:6]: done

File: src/lib/interfaces/thread.interface.ts (hunk 2 above)
Comment (vladutdinu) [reply src/lib/interfaces/thread.interface.ts:5]: done

File: src/lib/interfaces/thread.interface.ts (hunk 3 above)
Comment (vladutdinu) [reply src/lib/interfaces/thread.interface.ts:2]: done

File: src/lib/interfaces/thread.interface.ts (hunk 4 above)
Comment (vladutdinu) [reply src/lib/interfaces/thread.interface.ts:13]: done

File: src/routes/api/domains/+page.svelte (hunk 5 above)
Comment (vladutdinu) [reply src/routes/api/domains/+page.svelte:4]: done

File: src/routes/api/domains/+page.svelte (hunk 7 above)
Comment (vladutdinu) [reply src/routes/api/domains/+page.svelte:14]: done

File: src/routes/api/domains/[id]/subdomains/+page.svelte (hunk 8 above)
Comment (vladutdinu) [reply src/routes/api/domains/[id]/subdomains/+page.svelte:5]: ok

File: src/routes/api/domains/[id]/subdomains/+page.svelte (hunk 9 above)
Comment (vladutdinu) [reply src/routes/api/domains/[id]/subdomains/+page.svelte:8]: aoleu vrei controllere

File: src/routes/api/domains/[id]/subdomains/[id]/courses/+page.svelte (hunk 10 above)
Comment (vladutdinu) [reply src/routes/api/domains/[id]/subdomains/[id]/courses/+page.svelte:15]: da am inteles

File: src/routes/api/threads/+page.svelte (hunk 11 above)
Comment (vladutdinu) [reply src/routes/api/threads/+page.svelte:16]: am inteles ti am zis

**Review by smetytech — APPROVED** [review]

**Pull request merged**